core/qthelper.cpp

```diff
@@ -1653,3 +1653,33 @@ char *copy_qstring(const QString &s)
 {
 	return strdup(qPrintable(s));
 }
+
+// Generate a cylinder-renumber map for use when the n-th cylinder
+// of a dive with count cylinders is removed. It fills an int vector
+// with 0..n, -1, index..count-1. This probably should be moved
+// to the C-core, but using std::vector is simply more convenient.
+// This assumes that n < count!
+std::vector<int> get_cylinder_map_for_remove(int count, int n)
+{
+	// 1) Fill mapping[0]..mapping[index-1] with 0..index-1
+	// 2) Set mapping[index] to -1
+	// 3) Fill mapping[index+1]..mapping[count-1] with index..count-2
```

Comment:
again the references to `index` even thought that's nowhere in the code
And... instead of explaining what the code does, it would be even better to explain WHY that's what you do here (I think I have reached the point in my `std::` learning journey that I can actually read and understand that code... yay me)

Reply:
Replaced "index" by "n". Added a small comment to the function header what this does. Note that all this cylinder-remapping is not from me.